examples/dfsdemo/Makefile

```diff
@@ -9,6 +9,7 @@ OBJS = $(BUILD_DIR)/dfsdemo.o
 dfsdemo.z64: N64_ROM_TITLE = "DragonFS Demo"
 dfsdemo.z64: $(BUILD_DIR)/dfsdemo.dfs
 
+$(BUILD_DIR)/dfsdemo.dfs: filesystem/a_directory/a_file.txt
 $(BUILD_DIR)/dfsdemo.dfs: $(wildcard filesystem/*)
```

Comment:
In my opinion, the usage of `wildcard` here is a Makefile anti-pattern: `$(BUILD_DIR)/dfsdemo.dfs` does not get re-built when a file in a subdirectory of `filesystem` is touched because wildcard is not recursive.

I didn't want to touch that as part of fixing this issue, but I did want to make sure at least one of our examples exercised the edge case that this PR addresses.